census_view.cpp

```diff
@@ -1415,7 +2367,19 @@ void CensusView::UponColumnWidthFixed(wxCommandEvent&)
     Update();
 }
 
-void CensusView::UponRightClick(wxDataViewEvent& e)
+void CensusGridView::UponColumnWidthFixed(wxCommandEvent&)
+{
+    autosize_columns_ = false;
+
+    wxWindowUpdateLocker u(grid_window_);
+    for(int j = 0; j < bourn_cast<int>(grid_window_->GetNumberCols()); ++j)
+        {
+        grid_window_->SetColSize(j, wxCOL_WIDTH_DEFAULT);
+        }
+    Update();
```

Comment:
Calling `Update()` from inside `wxWindowUpdateLocker` looks strange. Is this really necessary/works as intended?

Reply:
I'm sorry, I'm afraid I was very wrong here. Reading the code now, I see that this is our own `Update()`, which has nothing to do with `wxWindow::Update()` (which is confusing on its own, but it's a different issue). And as our `Update()` also uses `wxWindowUpdateLocker`, this means that with the new version we're going to freeze/thaw the window twice, while without the separate scope we'd do it only once (it's safe to nest `wxWindowUpdateLocker`s into each other).

So this actually needs to be reverted, I think.